### PR TITLE
use VERSION file to specify app version

### PR DESCRIPTION
**Pull request by mniestroj**
Specify application version using `VERSION` file in application root directory. This is a standard mechanism provided by Zephyr.

`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` is automatically set to proper version according to `VERSION` file contents.

### CODE REVIEW

**Review by szczys — APPROVED** [review]
LGTM

With apologies to @cdwilson who suggested this change back in #66. Mea culpa.

File: src/main.c [hunk 1]
```diff
@@ -33,8 +34,9 @@ LOG_MODULE_REGISTER(golioth_rd_template, LOG_LEVEL_DBG);
 #include <modem/modem_info.h>
 #endif
 
-/* Current firmware version; update in prj.conf or via build argument */
-static const char *_current_version = CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION;
+// Current firmware version; update in VERSION
+static const char *_current_version =
+    STRINGIFY(APP_VERSION_MAJOR) "." STRINGIFY(APP_VERSION_MINOR) "." STRINGIFY(APP_PATCHLEVEL);
```
Comment (cdwilson) [inline src/main.c:39]: Should it be mentioned somewhere that `VERSION_TWEAK` and `EXTRAVERSION` fields are not used? I'm not even sure if these fields are supported by Golioth at all given that they are not supported in `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`. Just thinking that somebody might try to populate these in their fork and be confused as to why they aren't being picked up.

**Review by cdwilson — APPROVED** [review]
Also, FYI I saw this gotcha but haven't had a chance to test it myself yet: https://github.com/zephyrproject-rtos/zephyr/issues/69443

File: src/main.c (hunk 1 above)
Comment (mniestroj) [reply src/main.c:39]: I'm curious myself if `VERSION_TWEAK` would be  supported by Golioth. The reason I haven't used it is to keep the same behavior and just use 3 components/fields in the version string.

Comment (cdwilson) [reply src/main.c:39]: Oh, interesting, back when I originally looked at this, I think MCUboot only supported `APP_VERSION_MAJOR`, `APP_VERSION_MINOR`, and `APP_PATCHLEVEL`). Looks like MCUboot [now supports](https://github.com/nrfconnect/sdk-zephyr/blob/36fbc3ffc636fae0291b5d83ae6c59b8a814958d/modules/Kconfig.mcuboot#L106) the `VERSION_TWEAK` field?

Comment (mniestroj) [reply src/main.c:39]: MCUboot supported 4th component for very long time, maybe even from the start of the project. Maybe APP_VERSION infrastructure was not exposing that yet before. Also, we did not use it in Golioth SDK.

Comment (cdwilson) [reply src/main.c:39]: You're right. Maybe the original APP_VERSION implementation didn't include it (or I may have just misunderstood how it worked). LMK if you end up testing VERSION_TWEAK with Golioth, curious to know if it's supported.

**Pull request merged**